File: Problem206/Solution.go
package main

type ListNode struct {
	Val  int
	Next *ListNode
}

func reverseList(head *ListNode) *ListNode {
	if head == nil || head.Next == nil {
		return head
	}
	node := head.Next
	head.Next = nil
	reverse = reverseList(node)
	node.Next = head
	return reverse
}

File: Problem206/Solution.java
class Solution {
    public ListNode reverseList(ListNode head) {
        if (head == null || head.next == null) {
            return head;
        }

        ListNode next = head.next;
        ListNode reverse = reverseList(head.next);

        next.next = head;
        head.next = null;

        return reverse;
    }
}